integr8ly/nodejs-messaging-work-queue
# INTYL-3038 UI Update

## Description
To improve the overall user experience of the solution web explorer, the UI of the Node Messaging app from walkthrough 1 should be updated

This includes:

* Centering the layout
* Moving over to the Patternfly framework
* Toast notifications for a snappy UI feel 
* Any other design changes to improve look & feel of the app 

## Verification
**Note: The following steps uses an integreatly workshop provisioned on RHPDS. Additional steps for verification on a clean Openshift 3 or 4 cluster are required regarding integeattly installation that are not noted here. General verification steps remain the same, with differences in namespaces**
* On RHPDS, provision an Integreatly Workshop
* Browse to solution web explorer after installation and go through walkthrough 1 to integrate the CRUD app with the node js messaging app
* Login into the cluster via oc client
* Change project to the tutorial namespace
  * `oc project admin-tutorial-xxxx`
* Edit the depolyment config of the node js messaging
  * `oc edit dc nodejs-messaging-work-queue-frontend `
* Update the container image to point to the following quay.io image
  * `quay.io/kevfan/nodejs-messaging-work-queue:INTLY-3038-UI-Update`
* Wait for node messaging container to to redeployed 
  * `oc get pods -o`
* Visit/Refresh the node js messaging app url
* Ensure the application now uses the updated UI
* Verify fruit can still be created via the node messaging app
* Verify a success notification is shown on successful fruit creation
* Verify a error notification is shown when trying to creating fruit without any input to form field
* Verify there is now a footer with the following content:
```
Basic messaging application that represents a newer system using a queue.
```

## Screenshots
![Screenshot from 2019-09-13 12-12-18](https://user-images.githubusercontent.com/24636860/64858685-fb4d7b00-d61f-11e9-8877-c836d21090d7.png)
![Screenshot from 2019-09-13 12-13-11](https://user-images.githubusercontent.com/24636860/64858686-fb4d7b00-d61f-11e9-8fb1-7f9377955569.png)
![Screenshot from 2019-09-13 12-13-44](https://user-images.githubusercontent.com/24636860/64858687-fb4d7b00-d61f-11e9-9c5b-7b9c48bfd2a3.png)
